src/pages/Auth/RegisterStaff.tsx

```diff
@@ -0,0 +1,152 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { Link, useNavigate } from "react-router-dom";
+import * as Yup from "yup";
+import { useAuth } from "../../AuthContext";
+import FormError from "./FormError";
+
+interface FormValues {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+  programs: string;
+  counselor: boolean;
+  admin: boolean;
+}
+
+const schema = Yup.object().shape({
+  firstName: Yup.string().required("First Name is required"),
+  lastName: Yup.string().required("Last Name is required"),
+  email: Yup.string().email("Invalid email address").required("Email is required"),
+  password: Yup.string().min(5, "Password must be at least 5 characters").required("Password is required"),
+  confirmPassword: Yup.string().oneOf([Yup.ref("password")], "Passwords do not match").required("Confirm password is required"),
+  programs: Yup.string().required("Program selection is required"),
+  counselor: Yup.boolean(),
+  admin: Yup.boolean()
+});
+
+// add object for firebase uid here
+
+const RegisterStaff: React.FC = () => {
+  const { registerUser, currentUser } = useAuth();
+  const [role, setRole] = useState("");
+  const [uid, setUid] = useState(""); // State variable for storing UID
+
+  const handleRoleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRole(event.target.value);
+  };
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (currentUser) {
+      navigate("/");
+    }
+  }, [currentUser, navigate]);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<FormValues>({
+    resolver: yupResolver(schema),
+  });
+
+  const [error, setError] = useState<string>("");
+
+  const onSubmit = async (values: FormValues) => {
+    try {
+      values.counselor = role === "counselor";
+      values.admin = role === "admin";
+      setError("");
+      console.log("User values: ", values);
+
+      const name = values.firstName + " " + values.lastName;
+      const firebaseUID = await registerUser(name, values.email, values.password);
+      setUid(firebaseUID);
+
+      console.log("User id is: ", firebaseUID);
+      navigate("/"); // Redirect to home page
+    } catch (err: any) {
+      setError(err.message);
```

Comment:
might be useful to create an enum class for firebase errors to send more descriptive information. what do you think @lunstb?

Reply:
I don't have much experience working with firebase, I would think that just printing this out would be enough